CRISPResso2/CRISPRessoShared.py

```diff
@@ -1968,53 +1969,524 @@ def check_custom_config(args):
+    highReadsAlignedToAlternateAmplicon.safety(aln_stats['N_TOT_READS'], crispresso2_info['results']['alignment_stats']['counts_total'])
+
+    lowRatioOfModsInWindowToOut = LowRatioOfModsInWindowToOutGuardRail(messageHandler, guardrails['minRatioOfModsInToOut'])
+    lowRatioOfModsInWindowToOut.safety(aln_stats['N_MODS_IN_WINDOW'], aln_stats['N_MODS_OUTSIDE_WINDOW'])
+
+    highRateOfModificationAtEndsGuardRail = HighRateOfModificationAtEndsGuardRail(messageHandler, guardrails['modificationsAtEnds'])
+    highRateOfModificationAtEndsGuardRail.safety((aln_stats['N_CACHED_ALN'] + aln_stats['N_COMPUTED_ALN']), aln_stats['N_READS_IRREGULAR_ENDS'])
+
+    highRateOfSubstitutionsOutsideWindowGuardRail = HighRateOfSubstitutionsOutsideWindowGuardRail(messageHandler, guardrails['outsideWindowMaxSubRate'])
+    highRateOfSubstitutionsOutsideWindowGuardRail.safety(aln_stats['N_GLOBAL_SUBS'], aln_stats['N_SUBS_OUTSIDE_WINDOW'])
+
+    highRateOfSubstitutions = HighRateOfSubstitutionsGuardRail(messageHandler, guardrails['maxRateOfSubs'])
+    highRateOfSubstitutions.safety(aln_stats['N_MODS_IN_WINDOW'], aln_stats['N_MODS_OUTSIDE_WINDOW'], aln_stats['N_GLOBAL_SUBS'])
+
+    shortAmpliconSequence = ShortSequenceGuardRail(messageHandler, guardrails['amplicon_len'], 'amplicon')
+    shortAmpliconSequence.safety(amplicons)
+
+    shortGuideSequence = ShortSequenceGuardRail(messageHandler, guardrails['guide_len'], 'guide')
+    shortGuideSequence.safety(unique_guides)
+
+    longAmpliconShortReadsGuardRail = LongAmpliconShortReadsGuardRail(messageHandler, guardrails['ampliconToReadLen'])
+    longAmpliconShortReadsGuardRail.safety(amplicons, aln_stats['READ_LENGTH'])
+
+    return messageHandler.get_messages()
+
+
+class GuardRailMessageHandler:
+    """Class to handle message storage and display for guardrails"""
+    def __init__(self, logger):
+        """Create the message handler with an empty message array to collect html divs for the report
+
+        Parameters:
+        ------------
+        logger : logger
+            logger object used to display messages
+        """
+        self.logger = logger
+        self.messages = []
+
+    def display_warning(self, message):
+        """Send the message to the logger to be displayed
+
+        Parameters:
+        -----------
+        message : string
+            Related guardrail message
+        """
+        self.logger.warning(message)
+
+    def report_warning(self, message):
+        """Create and store the html message to display on the report
+
+        Parameters:
+        -----------
+        message : string
+            Related guardrail message
+        """
+        html_warning = '<div class="alert alert-danger"><strong>Guardrail Warning!</strong>{0}</div>'.format(message)
+        self.messages.append(html_warning)
+
+    def get_messages(self):
+        """Return the messages accumulated by the message handler"""
+        return self.messages
```

Comment:
Let's add the guardrails warnings to `crispresso2_info['results']['guardrails']` as a dictionary, where the key is the name of the guardrail class and the value is the message. A guardrail will only be added here if it is triggered.